lib/provisioning/Provisioner.js

```diff
@@ -0,0 +1,112 @@
+/*eslint no-invalid-this: 0*/ // eslint doesn't understand Promise.coroutine wrapping
+"use strict";
+var Promise = require("bluebird");
+var IrcRoom = require("../models/IrcRoom");
+var MatrixRoom = require("matrix-appservice-bridge").MatrixRoom;
+
+var log = require("../logging").get("Provisioner");
+
+function Provisioner(ircBridge) {
+    this._ircBridge = ircBridge;
+
+    let as = this._ircBridge.getAppServiceBridge().appService;
+
+    as.app.post("/_matrix/provision/link", (req, res) => {
+        return this.validate(req.body).then(
+            () => { res.json({}) },
+            (err) => { res.status(500).json({error: err.message}); return Promise.reject(err); }
+        ).then(
+            () => { return this.link(req.body) },
```

Comment:
It looks like you're returning the response on `:16` before actually seeing if `this.link` works? It's difficult to see what this is doing overall.